gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java

```diff
@@ -492,9 +494,25 @@ public GradeSaveResponse saveGrade(final Long assignmentId, final String student
 
 			// only convert if we had a previous value otherwise it will be out of sync
 			if (StringUtils.isNotBlank(oldGradeAdjusted)) {
+				// To check if our data is out of date, we first compare what we think
+				// is the latest saved score against score stored in the database. As the score
+				// is stored as points, we must convert this to a percentage. To be sure we're
+				// comparing apples with apples, we first determine the number of decimal places
+				// on the score, so the converted points-as-percentage is in the expected format. 
+				int numberOfDecimalPlaces = 0;
+				if (storedGradeAdjusted.indexOf(".") >= 0) {
```

Comment:
@steveswinsburg, I've had a go at ridding the GradebookNG of conversions when checking for concurrent edits, but it required jumping into the service to expose the raw `pointsEarned` value.  I also reworked the GradeSaveResponse from an enum to the class, so we can attach onto it the newly saved grade value.  This means we don't have to worry if our converted string-to-double is indeed what has been saved in the database, we only ever deal with the Double pointsEarned directly now.

Here's the branch: https://github.com/sakaiproject/sakai/compare/master...payten:3210-take-3

It feels like an improvement but I ended up touching more code than I expected.  What are your preferences for dealing with this one?  We can go with what we've got or happy to redo a PR for the take-3 attempt.